app/controllers/course/experience_points_records_controller.rb

```diff
@@ -0,0 +1,29 @@
+class Course::ExperiencePointsRecordsController < Course::ComponentController
+  load_resource :course_user, through: :course
+  load_and_authorize_resource :experience_points_record, through: :course_user,
+                                                         class: Course::ExperiencePointsRecord.name
+
+  def new # :nodoc:
+  end
+
+  def create # :nodoc:
+    if @experience_points_record.save
+      redirect_to course_course_user_experience_points_records_path(current_course, @course_user),
+                  success: t('.success')
+    else
+      render 'new'
+    end
+  end
+
+  def index # :nodoc:
```

Comment:
I happened to see this comment and went to take a look. Weiqing's earlier change specified the order for private controller methods.

Is the desired order `index`, `new`, `create`, `edit`, `update`, `destroy`? I got that from groups_controller. Where should `show` go?